### PR TITLE
alt-server: init at 0.0.5

### PR DESCRIPTION
**Pull request by max-amb**
## Description of changes

https://github.com/NyaMisty/AltServer-Linux

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by max-amb** [comment]
Sorry for the wait @drupol, was having problems with git!

File: pkgs/tools/misc/alt-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ stdenv
+, fetchurl
+, lib
+, autoPatchelfHook
+, avahi-compat}:
```
Comment (drupol) [inline pkgs/tools/misc/alt-server/default.nix:5]:
```suggestion
, avahi-compat
}:
```

Comment (max-amb) [reply pkgs/tools/misc/alt-server/default.nix]: Remade the commit!

File: pkgs/tools/misc/alt-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    chmod u+rx $out/bin/alt-server
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/NyaMisty/AltServer-Linux";
+    description = lib.mdDoc ''
+        AltServer for AltStore, but on-device. Requires root privileges as well as running a custom anisette server currently.
+    '';
```
Comment (drupol) [inline pkgs/tools/misc/alt-server/default.nix:37]: There's no need to have multiple lines here. One is enough.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/alt-server/default.nix (hunk 2 above)
Comment (max-amb) [reply pkgs/tools/misc/alt-server/default.nix]: Done!

**Review by drupol — CHANGES_REQUESTED** [review]
Looks like you're missing an entry in `all-packages.nix`.

**Review by drupol — APPROVED** [review]

**Comment by max-amb** [comment]
Thanks, sorry for being a pain!

**Comment by drupol** [comment]
Hey don't say that, we are all here to make mistakes and learn, so, don't worry :)

Keep PRs coming :)

**Pull request merged**